### ISSUE TITLE
Sandbox not restoring stubbed constructors in 1.17

### ISSUE THREAD

**Issue by neckardt** [sinonjs/sinon#851]
Our tests suddenly started failing after upgrading to the newest version of sinon. Stubbing objects that have a constructor while using the sandbox does not work properly -- sandbox.restore() does not restore the constructor. This issue does not occur in 1.16.

An example which replicates this issue:
```
describe.only('sandbox-test', function(){
  function MyClass(){}
  it('should not throw an error', function(){
    var myObject = new MyClass();
    var sandbox = sinon.sandbox.create();
    sandbox.stub(myObject);
    sandbox.restore();
    sandbox.stub(myObject);
  });
});
```

The error that occurs:
```
 sandbox-test
    1) should not throw an error


  0 passing (86ms)
  1 failing

  1) sandbox-test should not throw an error:
     TypeError: Attempted to wrap constructor which is already wrapped
      at checkWrappedMethod (node_modules/sinon/lib/sinon/util/core.js:81:29)
      at Object.wrapMethod (node_modules/sinon/lib/sinon/util/core.js:121:21)
      at stub (node_modules/sinon/lib/sinon/stub.js:66:26)
      at node_modules/sinon/lib/sinon/stub.js:59:25
      at node_modules/sinon/lib/sinon/walk.js:34:26
      at Array.forEach (native)
      at Object.walk (node_modules/sinon/lib/sinon/walk.js:33:45)
      at Object.stub (node_modules/sinon/lib/sinon/stub.js:52:23)
      at Object.stub (node_modules/sinon/lib/sinon/collection.js:106:49)
      at Context.<anonymous> (test/unit/sandbox-test.js:16:13)
```

**Comment by mroderick** [comment]
This looks like it's related to #847, #852

**Comment by traviskaufman** [comment]
Submitted PR which includes a fix. Also added a test in there that tests #852. Not sure this will fix #847 but I'm tackling that one next.

**Comment by mroderick** [comment]
Sinon.JS v1.17.1 is now on npm, the website will be updated soon.
